src/common/imagwebp.cpp

```diff
@@ -0,0 +1,365 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/imagwebp.cpp
+// Purpose:     wxImage WebP handler
+// Author:      Hermann Höhne
+// Created:     2024-03-08
+// Copyright:   (c) Hermann Höhne
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// based on code by Sylvain Bougnoux, Khral Steelforge and Markus Juergens
+// see https://forums.wxwidgets.org/viewtopic.php?t=39212
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if wxUSE_IMAGE && wxUSE_LIBWEBP
+
+#ifndef WX_PRECOMP
+    #include "wx/log.h"
+#endif
+
+#include "wx/imagwebp.h"
+
+#include <webp/demux.h>
+#include <webp/decode.h>
+#include <webp/encode.h>
+
+//-----------------------------------------------------------------------------
+// wxWEBPHandler
+//-----------------------------------------------------------------------------
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxWEBPHandler, wxImageHandler);
+
+#if wxUSE_STREAMS
+
+#include "wx/mstream.h"
+
+#include <memory>
+#include <functional>
+
+namespace
+{
+typedef std::unique_ptr<WebPDemuxer, std::function<void(WebPDemuxer*)>> WebPDemuxerPtr;
+typedef std::unique_ptr<WebPAnimDecoder, std::function<void(WebPAnimDecoder*)>> WebPAnimDecoderPtr;
+typedef std::unique_ptr<uint8_t, std::function<void(uint8_t*)>> WebPDecodeRGBAPtr;
+
+void rgbaToImage(wxImage* image, uint8_t* rgba)
+{
+    image->InitAlpha();
+    unsigned char* rgb = image->GetData();
+    unsigned char* alpha = image->GetAlpha();
+    size_t pixel_count = (size_t)image->GetWidth() * (size_t)image->GetHeight();
+    size_t rgba_index = 0, rgb_index = 0, alpha_index = 0;
+    for (size_t pixel_counter = 0; pixel_counter < pixel_count; pixel_counter++)
+    {
+        rgb[rgb_index++] = rgba[rgba_index++]; // R
+        rgb[rgb_index++] = rgba[rgba_index++]; // G
+        rgb[rgb_index++] = rgba[rgba_index++]; // B
+        alpha[alpha_index++] = rgba[rgba_index++]; // A
+    }
+}
+
+bool copyImageData(wxImage* to, wxImage& from)
```

Comment:
And this is already part of `wxImage`, it's called `Copy()` in it, so should be replaced with the call to this function.